jujugui/static/gui/src/app/components/profile/charm-list/test-charm-list.js

```diff
@@ -0,0 +1,124 @@
+/* Copyright (C) 2017 Canonical Ltd. */
+
+'use strict';
+
+var juju = {components: {}}; // eslint-disable-line no-unused-vars
+
+describe('Profile Charm List', function() {
+
+  const rawCharmData = `[{
+    "id": "cs:~hatch/precise/failtester-7",
+    "series": ["precise"],
+    "perm": {
+      "read": ["everyone", "hatch"],
+      "write": ["hatch"]
+    },
+    "name": "failtester"
+  }, {
+    "id": "cs:~hatch/xenial/ghost-3",
+    "series": ["xenial"],
+    "perm": {
+      "read": ["everyone", "hatch"],
+      "write": ["hatch"]
+    },
+    "name": "ghost"
+  }, {
+    "id": "cs:~hatch/privghost-1",
+    "series": ["xenial", "trusty"],
+    "perm": {
+      "read": ["hatch"],
+      "write": ["hatch"]
+    },
+    "name": "privghost"
+  }]`;
+
+  beforeAll(function(done) {
+    // By loading this file it adds the component to the juju components.
+    YUI().use('profile-charm-list', function() {
+      done();
+    });
+  });
+
+  function renderComponent(options={}) {
+    const charmstoreList = (user, cb) => {
+      assert.equal(user, 'hatch@external');
+      cb(null, JSON.parse(rawCharmData));
+    };
+    return jsTestUtils.shallowRender(
+      <juju.components.ProfileCharmList
+        addNotification={sinon.stub()}
+        baseURL="/gui/"
+        changeState={options.changeState || sinon.stub()}
+        charmstore={{
+          list: charmstoreList,
+          url: '/charmstore'
+        }}
+        user="hatch@external" />, true);
+  }
+
+  it('can render', () => {
```

Comment:
It seems like there are a number of other things you can test here e.g. the props updating with and without a new user, xhrs being cleared, error messages being display, navigating to charms etc.